backend/src/main/java/com/crm/service/impl/CarServiceImpl.java

```diff
@@ -78,8 +78,6 @@ private boolean hasVinIllegalCharacters(final String vin) {
 
     @Override
     public CarResponse addCar(CarRequest carRequest) {
-
-        CarTypeEntity carTypeEntity;
         String nameOfCarTypeEntity = carRequest.getCarTypeEntity().getName();
 
         if (carRepository.findByRegistrationNumberIgnoreCase(carRequest.getRegistrationNumber()).isPresent()) {
```

Comment:
This two if's can be moved to one private method I think.

checkIfCarAlreadyExist()

just a suggestion

Reply:
Good point, corrected